src/app/views/common/popups/PanelWrapper.tsx

```diff
@@ -1,109 +1,96 @@
-  }
+  };
 
-  const panelType = getPanelType();
+  const showBackButton =
+    title === 'Edit Scope' ||
+    title === 'Edit Collection' ||
+    title === 'Preview Permissions';
 
-  const onRenderFooterContent = (): JSX.Element | null => {
-    return renderFooter ? renderFooter() : null;
-  }
-
-  const showBackButton = title === 'Edit Scope' || title === 'Edit Collection' || title === 'Preview Permissions';
-
-  const onRenderHeader = (): JSX.Element => (
-    <div style={{ display: 'flex', alignItems: 'center' }}>
-      <IconButton
-        iconProps={{ iconName: 'Back' }}
-        ariaLabel={translateMessage('Back')}
-        onClick={() => dismissPopup()}
-        styles={{ root: { marginRight: 8 } }}
-      />
-      <span style={{ fontSize: theme.fonts.xLarge.fontSize, fontWeight: theme.fonts.xLarge.fontWeight }}>
-        {title}
-      </span>
-    </div>
-  );
+  const onOpenChange = (_ev: unknown, data: { open: boolean }) => {
+    if (!data.open) {
+      dismissPopup();
+    }
+  };
 
   return (
-    <div>
-      <Panel
-        isOpen={isOpen}
-        hasCloseButton={false}
-        type={panelType}
-        headerText={headerText.toString()}
-        isFooterAtBottom={true}
-        isBlocking={true}
-        isLightDismiss={false}
-        closeButtonAriaLabel='Close'
-        overlayProps={panelOverlayProps}
-        onRenderFooterContent={onRenderFooterContent}
-        onRenderHeader={showBackButton ? onRenderHeader: undefined}
-      >
-        <IconButton
-          styles={{
-            root: {
-              float: 'right',
-              zIndex: 1,
-              marginTop: -30
-            }
-          }}
-          iconProps={{ iconName: 'Cancel' }}
-          ariaLabel={translateMessage('Close')}
-          onClick={() => dismissPopup()}
-        />
-        <div>
-          {
-            <Suspense fallback={<Spinner />}>
-              <Component
-                {...popupsProps}
-                data={popupsProps.data || {}}
+    <Drawer
+      open={isOpen}
+      onOpenChange={onOpenChange}
+      position='end'
+      type='overlay'
+      size={getDrawerSize()}
+    >
+      <DrawerHeader>
+        {showBackButton && (
+          <Button
+            icon={<ArrowLeft24Regular />}
+            appearance='subtle'
+            onClick={() => dismissPopup()}
+            aria-label={translateMessage('Back')}
+          />
+        )}
+        <DrawerHeaderTitle action={
+          <Button
+            icon={<Dismiss24Regular />}
+            appearance='subtle'
+            onClick={() => dismissPopup()}
+            aria-label={translateMessage('Close')}
+          />
+
+        }>
+          {title || ''}
+        </DrawerHeaderTitle>
+      </DrawerHeader>
 
-                dismissPopup={() => dismissPopup()}
-                closePopup={(e: any) => closePopup(e)}
-              />
-            </Suspense>
-          }
-        </div>
+      <DrawerBody>
+        <Suspense fallback={<Spinner />}>
```

Comment:
I like this! I was thinking to make this the pattern to use for any content fetching components. In line with my other thought to remove the loading graph explorer loading page. What do you think?

Reply:
Yeah I agree with you on making this the pattern for content fetching components! 
What would you want to replace the graph explorer loading page with? The <Suspense> component as well?